File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/lifecycle_state.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LifecycleState(object):
    """
    The lifecycle state of execution.
    """

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "PENDING"
    STATUS_PENDING = "PENDING"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "QUEUED"
    STATUS_QUEUED = "QUEUED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "RUNNING"
    STATUS_RUNNING = "RUNNING"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "SKIPPED"
    STATUS_SKIPPED = "SKIPPED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "INTERNAL_ERROR"
    STATUS_INTERNAL_ERROR = "INTERNAL_ERROR"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "BLOCKED"
    STATUS_BLOCKED = "BLOCKED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "SUCCESS"
    STATUS_SUCCESS = "SUCCESS"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "FAILED"
    STATUS_FAILED = "FAILED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "CANCELING"
    STATUS_CANCELING = "CANCELING"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "CANCELED"
    STATUS_CANCELED = "CANCELED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "UPSTREAM_CANCELED"
    STATUS_UPSTREAM_CANCELED = "UPSTREAM_CANCELED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "UPSTREAM_FAILED"
    STATUS_UPSTREAM_FAILED = "UPSTREAM_FAILED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "EXCLUDED"
    STATUS_EXCLUDED = "EXCLUDED"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "TIMED_OUT"
    STATUS_TIMED_OUT = "TIMED_OUT"

    #: A constant which can be used with the status property of a LifecycleState.
    #: This constant has a value of "PAUSED_MAINTENANCE"
    STATUS_PAUSED_MAINTENANCE = "PAUSED_MAINTENANCE"

    def __init__(self, **kwargs):
        """
        Initializes a new LifecycleState object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param start_time:
            The value to assign to the start_time property of this LifecycleState.
        :type start_time: int

        :param end_time:
            The value to assign to the end_time property of this LifecycleState.
        :type end_time: int

        :param status:
            The value to assign to the status property of this LifecycleState.
            Allowed values for this property are: "PENDING", "QUEUED", "RUNNING", "SKIPPED", "INTERNAL_ERROR", "BLOCKED", "SUCCESS", "FAILED", "CANCELING", "CANCELED", "UPSTREAM_CANCELED", "UPSTREAM_FAILED", "EXCLUDED", "TIMED_OUT", "PAUSED_MAINTENANCE"
        :type status: str

        :param state_message:
            The value to assign to the state_message property of this LifecycleState.
        :type state_message: str

        :param error_trace:
            The value to assign to the error_trace property of this LifecycleState.
        :type error_trace: str

        :param is_user_canceled_or_timed_out:
            The value to assign to the is_user_canceled_or_timed_out property of this LifecycleState.
        :type is_user_canceled_or_timed_out: bool

        """
        self.swagger_types = {
            'start_time': 'int',
            'end_time': 'int',
            'status': 'str',
            'state_message': 'str',
            'error_trace': 'str',
            'is_user_canceled_or_timed_out': 'bool'
        }

        self.attribute_map = {
            'start_time': 'startTime',
            'end_time': 'endTime',
            'status': 'status',
            'state_message': 'stateMessage',
            'error_trace': 'errorTrace',
            'is_user_canceled_or_timed_out': 'isUserCanceledOrTimedOut'
        }

        self._start_time = None
        self._end_time = None
        self._status = None
        self._state_message = None
        self._error_trace = None
        self._is_user_canceled_or_timed_out = None

    @property
    def start_time(self):
        """
        Gets the start_time of this LifecycleState.
        The time at which the job execution started in epoch milliseconds.


        :return: The start_time of this LifecycleState.
        :rtype: int
        """
        return self._start_time

    @start_time.setter
    def start_time(self, start_time):
        """
        Sets the start_time of this LifecycleState.
        The time at which the job execution started in epoch milliseconds.


        :param start_time: The start_time of this LifecycleState.
        :type: int
        """
        self._start_time = start_time

    @property
    def end_time(self):
        """
        Gets the end_time of this LifecycleState.
        The time at which the job execution started in epoch milliseconds.


        :return: The end_time of this LifecycleState.
        :rtype: int
        """
        return self._end_time

    @end_time.setter
    def end_time(self, end_time):
        """
        Sets the end_time of this LifecycleState.
        The time at which the job execution started in epoch milliseconds.


        :param end_time: The end_time of this LifecycleState.
        :type: int
        """
        self._end_time = end_time

    @property
    def status(self):
        """
        Gets the status of this LifecycleState.
        Current state of execution.

        Allowed values for this property are: "PENDING", "QUEUED", "RUNNING", "SKIPPED", "INTERNAL_ERROR", "BLOCKED", "SUCCESS", "FAILED", "CANCELING", "CANCELED", "UPSTREAM_CANCELED", "UPSTREAM_FAILED", "EXCLUDED", "TIMED_OUT", "PAUSED_MAINTENANCE"


        :return: The status of this LifecycleState.
        :rtype: str
        """
        return self._status

    @status.setter
    def status(self, status):
        """
        Sets the status of this LifecycleState.
        Current state of execution.


        :param status: The status of this LifecycleState.
        :type: str
        """
        allowed_values = ["PENDING", "QUEUED", "RUNNING", "SKIPPED", "INTERNAL_ERROR", "BLOCKED", "SUCCESS", "FAILED", "CANCELING", "CANCELED", "UPSTREAM_CANCELED", "UPSTREAM_FAILED", "EXCLUDED", "TIMED_OUT", "PAUSED_MAINTENANCE"]
        if not value_allowed_none_or_none_sentinel(status, allowed_values):
            raise ValueError(
                "Invalid value for `status`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._status = status

    @property
    def state_message(self):
        """
        Gets the state_message of this LifecycleState.
        A descriptive message of the current state.


        :return: The state_message of this LifecycleState.
        :rtype: str
        """
        return self._state_message

    @state_message.setter
    def state_message(self, state_message):
        """
        Sets the state_message of this LifecycleState.
        A descriptive message of the current state.


        :param state_message: The state_message of this LifecycleState.
        :type: str
        """
        self._state_message = state_message

    @property
    def error_trace(self):
        """
        Gets the error_trace of this LifecycleState.
        If there was an error executing the run, this field contains any available stack traces.


        :return: The error_trace of this LifecycleState.
        :rtype: str
        """
        return self._error_trace

    @error_trace.setter
    def error_trace(self, error_trace):
        """
        Sets the error_trace of this LifecycleState.
        If there was an error executing the run, this field contains any available stack traces.


        :param error_trace: The error_trace of this LifecycleState.
        :type: str
        """
        self._error_trace = error_trace

    @property
    def is_user_canceled_or_timed_out(self):
        """
        Gets the is_user_canceled_or_timed_out of this LifecycleState.
        Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.


        :return: The is_user_canceled_or_timed_out of this LifecycleState.
        :rtype: bool
        """
        return self._is_user_canceled_or_timed_out

    @is_user_canceled_or_timed_out.setter
    def is_user_canceled_or_timed_out(self, is_user_canceled_or_timed_out):
        """
        Sets the is_user_canceled_or_timed_out of this LifecycleState.
        Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.


        :param is_user_canceled_or_timed_out: The is_user_canceled_or_timed_out of this LifecycleState.
        :type: bool
        """
        self._is_user_canceled_or_timed_out = is_user_canceled_or_timed_out

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/LifecycleState.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The lifecycle state of execution.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LifecycleState.Builder.class)

public final class LifecycleState  {
    @Deprecated
    @java.beans.ConstructorProperties({"startTime", "endTime", "status", "stateMessage", "errorTrace", "isUserCanceledOrTimedOut"})
    public LifecycleState(Long startTime, Long endTime, Status status, String stateMessage, String errorTrace, Boolean isUserCanceledOrTimedOut) {
        super();
        this.startTime = startTime;
        this.endTime = endTime;
        this.status = status;
        this.stateMessage = stateMessage;
        this.errorTrace = errorTrace;
        this.isUserCanceledOrTimedOut = isUserCanceledOrTimedOut;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("startTime")
        private Long startTime;

                /**
         * The time at which the job execution started in epoch milliseconds.
         * @param startTime the value to set
         * @return this builder
         **/
        

        public Builder startTime(Long startTime) {
        this.startTime = startTime;
        return this;
        }
            /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("endTime")
        private Long endTime;

                /**
         * The time at which the job execution started in epoch milliseconds.
         * @param endTime the value to set
         * @return this builder
         **/
        

        public Builder endTime(Long endTime) {
        this.endTime = endTime;
        return this;
        }
            /**
     * Current state of execution.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("status")
        private Status status;

                /**
         * Current state of execution.
         * @param status the value to set
         * @return this builder
         **/
        

        public Builder status(Status status) {
        this.status = status;
        return this;
        }
            /**
     * A descriptive message of the current state.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("stateMessage")
        private String stateMessage;

                /**
         * A descriptive message of the current state.
         * @param stateMessage the value to set
         * @return this builder
         **/
        

        public Builder stateMessage(String stateMessage) {
        this.stateMessage = stateMessage;
        return this;
        }
            /**
     * If there was an error executing the run, this field contains any available stack traces.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("errorTrace")
        private String errorTrace;

                /**
         * If there was an error executing the run, this field contains any available stack traces.
         * @param errorTrace the value to set
         * @return this builder
         **/
        

        public Builder errorTrace(String errorTrace) {
        this.errorTrace = errorTrace;
        return this;
        }
            /**
     * Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isUserCanceledOrTimedOut")
        private Boolean isUserCanceledOrTimedOut;

                /**
         * Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.
         * @param isUserCanceledOrTimedOut the value to set
         * @return this builder
         **/
        

        public Builder isUserCanceledOrTimedOut(Boolean isUserCanceledOrTimedOut) {
        this.isUserCanceledOrTimedOut = isUserCanceledOrTimedOut;
        return this;
        }


        public LifecycleState build() {
            LifecycleState model = new LifecycleState(this.startTime
                    , this.endTime
                    , this.status
                    , this.stateMessage
                    , this.errorTrace
                    , this.isUserCanceledOrTimedOut);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LifecycleState model) {
                this.startTime(model.getStartTime());
    this.endTime(model.getEndTime());
    this.status(model.getStatus());
    this.stateMessage(model.getStateMessage());
    this.errorTrace(model.getErrorTrace());
    this.isUserCanceledOrTimedOut(model.getIsUserCanceledOrTimedOut());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("startTime")
    private final Long startTime;

        /**
     * The time at which the job execution started in epoch milliseconds.
     * @return the value
     **/
    
    public Long getStartTime() {
        return startTime;
    }


        /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("endTime")
    private final Long endTime;

        /**
     * The time at which the job execution started in epoch milliseconds.
     * @return the value
     **/
    
    public Long getEndTime() {
        return endTime;
    }

    /**
     * Current state of execution.
     **/
    public enum Status {
        Pending("PENDING"),
        Queued("QUEUED"),
        Running("RUNNING"),
        Skipped("SKIPPED"),
        InternalError("INTERNAL_ERROR"),
        Blocked("BLOCKED"),
        Success("SUCCESS"),
        Failed("FAILED"),
        Canceling("CANCELING"),
        Canceled("CANCELED"),
        UpstreamCanceled("UPSTREAM_CANCELED"),
        UpstreamFailed("UPSTREAM_FAILED"),
        Excluded("EXCLUDED"),
        TimedOut("TIMED_OUT"),
        PausedMaintenance("PAUSED_MAINTENANCE"),
        ;

        

        private final String value;
        private static java.util.Map<String, Status> map;

        static {
            map = new java.util.HashMap<>();
            for (Status v : Status.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Status(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Status create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Status: " + key);
        }
    };
        /**
     * Current state of execution.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("status")
    private final Status status;

        /**
     * Current state of execution.
     * @return the value
     **/
    
    public Status getStatus() {
        return status;
    }


        /**
     * A descriptive message of the current state.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("stateMessage")
    private final String stateMessage;

        /**
     * A descriptive message of the current state.
     * @return the value
     **/
    
    public String getStateMessage() {
        return stateMessage;
    }


        /**
     * If there was an error executing the run, this field contains any available stack traces.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("errorTrace")
    private final String errorTrace;

        /**
     * If there was an error executing the run, this field contains any available stack traces.
     * @return the value
     **/
    
    public String getErrorTrace() {
        return errorTrace;
    }


        /**
     * Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isUserCanceledOrTimedOut")
    private final Boolean isUserCanceledOrTimedOut;

        /**
     * Set to true, if the job execution is canceled by the user or by the scheduler due to timeout.
     * @return the value
     **/
    
    public Boolean getIsUserCanceledOrTimedOut() {
        return isUserCanceledOrTimedOut;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LifecycleState(");
        sb.append("startTime=").append(String.valueOf(this.startTime));
        sb.append(", endTime=").append(String.valueOf(this.endTime));
        sb.append(", status=").append(String.valueOf(this.status));
        sb.append(", stateMessage=").append(String.valueOf(this.stateMessage));
        sb.append(", errorTrace=").append(String.valueOf(this.errorTrace));
        sb.append(", isUserCanceledOrTimedOut=").append(String.valueOf(this.isUserCanceledOrTimedOut));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LifecycleState)) {
            return false;
        }

        LifecycleState other = (LifecycleState) o;
        return java.util.Objects.equals(this.startTime, other.startTime) &&
            java.util.Objects.equals(this.endTime, other.endTime) &&
            java.util.Objects.equals(this.status, other.status) &&
            java.util.Objects.equals(this.stateMessage, other.stateMessage) &&
            java.util.Objects.equals(this.errorTrace, other.errorTrace) &&
            java.util.Objects.equals(this.isUserCanceledOrTimedOut, other.isUserCanceledOrTimedOut);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.startTime == null ? 43 : this.startTime.hashCode());
        result = (result * PRIME) + (this.endTime == null ? 43 : this.endTime.hashCode());
        result = (result * PRIME) + (this.status == null ? 43 : this.status.hashCode());
        result = (result * PRIME) + (this.stateMessage == null ? 43 : this.stateMessage.hashCode());
        result = (result * PRIME) + (this.errorTrace == null ? 43 : this.errorTrace.hashCode());
        result = (result * PRIME) + (this.isUserCanceledOrTimedOut == null ? 43 : this.isUserCanceledOrTimedOut.hashCode());
        return result;
    }


}
